[core-kit] dev-util/ninja Add autogen and template files for ninja package

org-tekeli-borisp opened the pull request:
Introduce an autogen YAML and template for the `dev-util/ninja` package configuration. This integrates the necessary build, test, and install processes, including support for optional features like documentation, Emacs, and Vim syntax enhancements.

(cherry picked from commit 0f34a7482448c92ffe303315d0578204c8da982f)

org-tekeli-borisp committed:
[core-kit] dev-util/ninja Add autogen and template files for ninja package

Introduce an autogen YAML and template for the `dev-util/ninja` package configuration. This integrates the necessary build, test, and install processes, including support for optional features like documentation, Emacs, and Vim syntax enhancements.

(cherry picked from commit 0f34a7482448c92ffe303315d0578204c8da982f)

geaaru commented on core-kit/curated/dev-util/ninja/templates/ninja.tmpl:
```diff
@@ -0,0 +1,121 @@
+
+KEYWORDS="*"
+SLOT="0"
+IUSE="doc emacs test vim-syntax"
```
I think that we can remove *test* USE flag and related code

org-tekeli-borisp replied:
@geaaru  src_test() too?

geaaru replied:
yeah, exactly. Less unused code it's better.
Personally, I prefer to organize testing with a separated workflow in the future.

org-tekeli-borisp committed:
[core-kit] dev-util/ninja Remove support for 'test' USE flag in ninja template

The 'test' USE flag and related functionality have been removed, including dependencies on gtest and the `src_test` function. This simplifies the template and removes unnecessary test-related logic.
